### PR TITLE
Fixes #322

### LINKED ISSUE #322: ColumnMeaning detection is a bit too eager
The detection of ColumnMeaning sometimes has false positives, i.e. anything ending in ID becomes a key column, and "phone no" becomes _Street Number_

### PR DESCRIPTION
**Pull request by kaspersorensen**
Fixes the "too eager column meaning mapping". See #322.

### CODE REVIEW

File: api/src/main/java/org/datacleaner/metadata/ColumnMeaning.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
 
     OTHER("Disregard", "Other", "Nothing"),
 
-    KEY_PRIMARY("Primary Key", "Record ID", "ID", "Identifier", "key", "PKID", "Record key"),
+    KEY_PRIMARY("Primary Key", "ID", "Record ID", "Identifier", "key", "PKID", "Record key"),
```
Comment (LosD) [inline api/src/main/java/org/datacleaner/metadata/ColumnMeaning.java]: Why the switch-around? Optimization? (not a problem, just curious)

Comment (kaspersorensen) [inline api/src/main/java/org/datacleaner/metadata/ColumnMeaning.java]: Hmm coincidence I think. I first tried another approach but changed my mind. Think this is a left-over of that attempt :-)

File: api/src/main/java/org/datacleaner/metadata/ColumnMeaning.java [hunk 2]
```diff
@@ -63,8 +63,8 @@
 
     COMPANY_REGISTRATION_NUMBER("Company registration number", "Reg number"),
 
-    COMPANY_NAME("Company name", "Company", "Organization", "Business", "Organisation", "Employer", "Firm",
-            "Workplace", "Works at", "Vendor", "Supplier", "Customer", "Customer name"),
+    COMPANY_NAME("Company name", "Company", "Organization", "Business", "Organisation name", "Employer", "Firm",
+            "Workplace", "Works at", "Vendor name", "Supplier name", "Customer name"),
```
Comment (LosD) [inline api/src/main/java/org/datacleaner/metadata/ColumnMeaning.java]: Now that we don't tokenize column meanings isn't it a problem that _Supplier_ and _Vendor_ are only there in combination with name?

**Comment by LosD** [comment]
Single worry, but looks good. We might want to make sure that other uses of the matcher isn't relying on the previous behavior (esp. tokenization).

File: api/src/main/java/org/datacleaner/metadata/ColumnMeaning.java (hunk 2 above)
Comment (kaspersorensen) [inline api/src/main/java/org/datacleaner/metadata/ColumnMeaning.java]: Yes, I'll add that.

**Pull request merged**